### PR TITLE
fix(libpod): keep '=' in a volume subpath

### PR DESCRIPTION
**Pull request by Atishyy27**
The subpath mount option is parsed with `strings.Split(o, "=")[1]`, so it keeps only the text between the first and second `=`. A subpath that contains `=` gets cut short and the container mounts a different directory than the one that was asked for.

```
--mount type=volume,src=v,dst=/mnt,subpath=/opt/a=b/c   mounts /opt/a
```

The existing subpath e2e test now uses a path with an `=` in it.

### CODE REVIEW

File: test/e2e/run_volume_test.go [hunk 1]
```diff
@@ -1127,6 +1127,25 @@ RUN chmod 755 /test1 /test2 /test3`, ALPINE)
 		Expect(checkCtr.OutputToString()).To(ContainSubstring("bar"))
 	})
 
+	It("podman run --mount type=volume,subpath= with an equals sign in the path", func() {
```
Comment (Luap99) [inline test/e2e/run_volume_test.go]: please add this to the exiting test, there is no need to duplicate this with two different paths. If we test s ginle path with `=` that is good enough

**Review by Luap99 — CHANGES_REQUESTED** [review]
please follow our LLM  policy https://github.com/podman-container-tools/community/blob/main/LLM_POLICY.md and contributing guide https://github.com/podman-container-tools/community/blob/main/CONTRIBUTING.md

The commit message as is is clearly AI generated and contains way to may unrelated details, submitting untested changes is generally a great way to waste our time.
The first two paragraphs would be enough, the rest is just useless noise.

File: test/e2e/run_volume_test.go [hunk 2]
```diff
@@ -1127,6 +1127,25 @@ RUN chmod 755 /test1 /test2 /test3`, ALPINE)
...
+		volName := "testvol"
+		mkvol := podmanTest.Podman([]string{"volume", "create", volName})
```
Comment (Honny1) [inline test/e2e/run_volume_test.go]: Please use `podmanTest.PodmanExitCleanly`

**Review by Luap99 — APPROVED** [review]
LGTM

**Review by Honny1 — APPROVED** [review]
LGTM, Thanks!
